cmd/branchProtection.go

```diff
@@ -232,34 +232,59 @@ func (verifier BranchProtectionVerifier) getRequiredChecks(requireSonar bool) ([
 	return result, nil
 }
 
-func (verifier BranchProtectionVerifier) getChecksForWorkflow(workflowFilePath *string) ([]string, error) {
+func (verifier BranchProtectionVerifier) getChecksForWorkflow(workflowFilePath *string, defaultBranch string) ([]string, error) {
 	content, err := verifier.downloadFile(*workflowFilePath)
 	if err != nil {
 		return nil, err
 	}
-	return verifier.getChecksForWorkflowContent(content, workflowFilePath), nil
+	return verifier.getChecksForWorkflowContent(content, workflowFilePath, defaultBranch), nil
 }
 
-func (verifier BranchProtectionVerifier) getChecksForWorkflowContent(content string, fileName *string) []string {
+func (verifier BranchProtectionVerifier) getChecksForWorkflowContent(content string, fileName *string, defaultBranch string) []string {
 	fileUrl := fmt.Sprintf("https://github.com/exasol/%s/blob/%s/%s", verifier.repoName, verifier.getRepo().GetDefaultBranch(), *fileName)
 	workflow, err := WorkflowDefinitionParser{}.ParseWorkflowDefinition(content)
 	if err != nil {
-		switch err := err.(type) {
-		case ValidationError:
-			fmt.Printf("%vValidation Error for '%v': %v %v\n", consoleColorRed, fileUrl, err.Error(), consoleColorReset)
-			os.Exit(1)
-		default:
-			fmt.Printf("%vWarning: Failed to parse workflow definition '%v'. Probably you use some advanced matrix build features there. Github-keeper will not add the checks from this workflow to the branch protection. Please add them manually. %v\n", consoleColorYellow, fileUrl, consoleColorReset)
-			var emptyResult []string
-			return emptyResult
-		}
+		handleParseError(err, fileUrl)
+		return nil
 	}
-	hasWorkflowPushOrPrTrigger := hasWorkflowPushOrPrTrigger(workflow.Trigger)
+	hasWorkflowPushOrPrTrigger := checkIfProtectionNeeded(workflow.Trigger, defaultBranch)
 	if hasWorkflowPushOrPrTrigger {
-		return workflow.JobsNames
+		jobNames, err := workflow.GetJobNames()
+		if err != nil {
+			handleParseError(err, fileUrl)
+			return nil
+		} else {
+			return jobNames
+		}
+	}
+	return nil
+}
+
+func handleParseError(err error, fileUrl string) {
+	switch err := err.(type) {
+	case ValidationError:
+		fmt.Printf("%vValidation Error for '%v': %v %v\n", consoleColorRed, fileUrl, err.Error(), consoleColorReset)
+		os.Exit(1)
+	default:
+		printParseFailedWarning(fileUrl)
+		return
+	}
+}
+
+func printParseFailedWarning(fileUrl string) {
+	fmt.Printf("%vWarning: Failed to parse workflow definition '%v'. Probably you use some advanced matrix build features there. Github-keeper will not add the checks from this workflow to the branch protection. Please add them manually. %v\n", consoleColorYellow, fileUrl, consoleColorReset)
+}
+
+func checkIfProtectionNeeded(triggers *TriggerDefinition, defaultBranch string) bool {
+	if triggers.TriggerOnPr || triggers.TriggerOnPushToAnyBranch {
+		return true
+	}
+	for _, branch := range triggers.TriggerOnPushToBranches {
+		if branch == defaultBranch {
+			return true
```

Comment:
I think, this is an error, because this means the workflow is executed only on push. Or, is the intention here, if this workflow runs on the default branch it should also run on feature branch. I think, this is not correct, in all cases.

Reply:
that's true